### PR TITLE
[release-2.2] Backport gatekeeper test updates

### PR DESCRIPTION
**Pull request by JustinKuli**
The changes to these tests should allow them to function correctly for
both the "old" and "new" versions of the gatekeeper and operator. Those
projects adjusted some details in their deployments and messages.

Refs:
 - https://github.com/open-cluster-management/backlog/issues/14914

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

This PR also updates the release-2.3 branch tests / make targets to use the `latest-2.3` images.

### CODE REVIEW

**Comment by JustinKuli** [comment]
/hold

need CICD approval

**Comment by JustinKuli** [comment]
For the equivalent PR for the 2.3 branch, I was able to update those tests to use the latest-2.3 images. But there are no latest-2.2 images for this set... we might just need to merge this as-is and look for the updates in the 2.2 canary tests.

**Comment by dhaiducek** [comment]
Sounds good--I have code ready to go to create `latest-2.2` tags, but I'm not sure it's worth the effort...

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**
